File: wisefy/src/androidTest/java/com/isupatches/wisefy/main/IsDeviceConnectedToMobileNetworkTests.java
package com.isupatches.wisefy.main;

import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

import com.isupatches.wisefy.AbstractBaseAndroidJUnit4TestClass;
import com.isupatches.wisefy.WiseFy;

import org.junit.Test;

/**
 * Tests the ability to determine if a device is connected to a mobile network.
 *
 * @see WiseFy#isDeviceConnectedToMobileNetwork()
 */
public class IsDeviceConnectedToMobileNetworkTests extends AbstractBaseAndroidJUnit4TestClass {

  public IsDeviceConnectedToMobileNetworkTests() {
    // No-op
  }

  @Test
  public void failure_prechecks() {
    getMockWiseFyPrechecksUtil().isDeviceConnectedToMobileNetwork_failure();
    assertFalse(getWiseFy().isDeviceConnectedToMobileNetwork());
  }

  @Test
  public void failure() {
    getMockWiseFyConnectionUtil().isNetworkConnectedAndMatchesType(false);
    assertFalse(getWiseFy().isDeviceConnectedToMobileNetwork());
  }

  @Test
  public void success() {
    getMockWiseFyPrerequisitesUtil().activeNetwork();
    getMockWiseFyConnectionUtil().isNetworkConnectedAndMatchesType(true);
    assertTrue(getWiseFy().isDeviceConnectedToMobileNetwork());
  }
}

File: wisefy/src/androidTest/java/com/isupatches/wisefy/IsDeviceConnectedToMobileNetworkTests.kt
package com.isupatches.wisefy

import com.isupatches.wisefy.internal.base.BaseInstrumentationTest
import org.junit.Assert.assertFalse
import org.junit.Assert.assertTrue
import org.junit.Test

/**
 * Tests the ability to determine if a device is connected to a mobile network.
 *
 * @author Patches
 * @since 3.0
 */
internal class IsDeviceConnectedToMobileNetworkTests : BaseInstrumentationTest() {

    @Test
    fun failure_prechecks() {
        mockWiseFyPrechecksUtil.isDeviceConnectedToMobileNetwork_failure()
        assertFalse(wisefy.isDeviceConnectedToMobileNetwork())
    }

    @Test
    fun failure() {
        mockWiseFyPrechecksUtil.isDeviceConnectedToMobileNetwork_success()
        mockWiseFyConnectionUtil.isDeviceConnectedToMobileNetwork(false)
        assertFalse(wisefy.isDeviceConnectedToMobileNetwork())
    }

    @Test
    fun success() {
        mockWiseFyPrechecksUtil.isDeviceConnectedToMobileNetwork_success()
        mockNetworkUtil.activeNetwork()
        mockWiseFyConnectionUtil.isDeviceConnectedToMobileNetwork(true)
        assertTrue(wisefy.isDeviceConnectedToMobileNetwork())
    }
}
